.github/workflows/CI.yaml

```diff
@@ -163,3 +163,26 @@ jobs:
 
       - name: Upload coverage to Codecov
         uses: codecov/codecov-action@v1
+
+  latest-dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up latest stable Python 3
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install tomli
+          eval "python -m pip install $(./bin/latest_dependencies.py)"
```

Comment:
Since this line is the crucial part of this job, how about we turn this line into its own step (separate from pip/tomli/pytest/etc. updates) such that when reviewing it will be faster to check which versions were being installed?

Reply:
"this line" referring to line 180 only...

Reply:
I'll change the minimum dependency check too